packages/libs/chainweb-node-client/src/interfaces/PactAPI.ts

```diff
@@ -0,0 +1,303 @@
+import type {
+  ChainId,
+  IMetaData,
+  ICommand,
+  IUnsignedCommand,
+  PactValue,
+  IBase64Url,
+  IPactExec,
+  IPactEvent,
+  SPVProof,
+} from '@kadena/types';
+
+/**
+ * The result of a pact execution when no errors occur.
+ * The data will be the value returned or generated by the pact code.
+ *
+ * @alpha
+ */
+interface IPactResultSuccess {
+  status: 'success';
+  data: PactValue;
+}
+
+/**
+ * The result of a pact execution when an error occurs.
+ * If the result is from a committed transaction from the /send endpoint, the error will only list the type of error that occured.
+ * If the result is from the /local endpoint, the error will be an object.
+ *
+ * @alpha
+ */
+interface IPactResultError {
+  status: 'failure';
+  error: object;
+}
+
+/**
+ * Platform-specific information on the block that executed a transaction.
+ *
+ * @param blockHash - Block hash of the block containing the transaction.
+ * @param blockTime - POSIX time when the block was mined.
+ * @param blockHeight - Block height of the block.
+ * @param prevBlockHash - Parent Block hash of the containing block.
+ * @param publicMeta - Platform-specific data provided by the request.
+ *
+ *
+ */
+// @TODO Add `publicMeta` to Open API spec.
+interface IChainwebResponseMetaData {
+  blockHash: string;
+  blockTime: number;
+  blockHeight: number;
+  prevBlockHash: string;
+  publicMeta?: IMetaData;
+}
+
+// TODO: Move Chainweb Specific Types
+/**
+ * Stringified Chainweb chain numbers.
+ * @alpha
+ */
+export type ChainwebChainId =
+  | '0'
+  | '1'
+  | '2'
+  | '3'
+  | '4'
+  | '5'
+  | '6'
+  | '7'
+  | '8'
+  | '9'
+  | '10'
+  | '11'
+  | '12'
+  | '13'
+  | '14'
+  | '15'
+  | '16'
+  | '17'
+  | '18'
+  | '19';
+
+/**
+ * Different Chainweb network versions.
+ * @alpha
+ */
+export type ChainwebNetworkId =
+  | 'mainnet01'
+  | 'testnet04'
+  | 'development'
+  | undefined;
```

Comment:
Should there be an `undefined` `ChainwebNetworkId`?

Reply:
No, we can remove `undefined`.